tests/test_context.py

```diff
@@ -0,0 +1,46 @@
+# Source pytest module
+# Copyright (C) 2020  Ngô Ngọc Đức Huy
+#
+# This file is part of palace.
+#
+# palace is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# palace is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with palace.  If not, see <https://www.gnu.org/licenses/>.
+
+"""This pytest module tries to test the correctness of the class Context."""
+
+from palace import Context
+
+
+def test_with_context(device):
+    """Test if `with` can be used to start a context
+    and is destroyed properly.
+    """
+    context = Context(device)
+    assert Context.current_context() is None
+    with context:
+        assert Context.current_context() is context
+    assert Context.current_context() is None
+
+
+def test_nested_context(device):
+    """Test if the context manager returns to the
+    previous context."""
+    ctx_1 = Context(device)
+    ctx_2 = Context(device)
+    assert Context.current_context() is None
+    with ctx_1:
+        assert Context.current_context() is ctx_1
+        with ctx_2:
+            assert Context.current_context() is ctx_2
+        assert Context.current_context() is ctx_1
```

Comment:
With context manager reconstructed in `current_context` (again, module-level function please), we'll have to rely on alure's API for equal comparison instead of asserting for the exact instance.